types/react-intl/index.d.ts

```diff
@@ -11,6 +11,10 @@
 declare namespace ReactIntl {
 
     interface Values {
+        [key: string]: string | JSX.Element;
```

Comment:
This interface is also used in `formatMessage` method which accepts only string values. JSX elements are only acceptable when using `FormattedMessage` you should create separate interface for that  or generic interface. This is duplicate of #15499 and look at the comments there what should be done.